.gitmodules

```diff
@@ -0,0 +1,3 @@
+[submodule "libraw"]
```

Comment:
I'm not sure it's a good idea to make this a submodule. I see the point of it, but given that this repo is not much maintained, the submodule will inevitably point to an outdated commit and I definitely don't want to merge "Bump submodule" PRs all the time.
Also, by changing the expected folder layout this will break anyone using this repo, or least the next time they update. For example https://github.com/microsoft/vcpkg/blob/master/ports/libraw/portfile.cmake.
I think given the special state of this repo it would make sense to maintain backwards compatibility while still supporting your use case of consuming it via `add_subdirectory`. This could be done by making the libraw source location a cache variable which defaults to CMAKE_CURRENT_SOURCE_DIR (in line with the existing CMake script semantics). In your case, you would set the source dir to something else, most likely a folder next to where this repo was cloned. This way you can make this and the libraw repo two submodules and still consume it via add_subdirectory.
Would that work?

Reply:
I think this works, I added the variable and reverted the other changes